src/lib/y2storage/proposal/space_maker.rb

```diff
@@ -284,8 +284,8 @@ def free_spaces(graph, disk = nil)
       #
       # @return [Array<Dasd, Disk>]
       def disks_for(devicegraph, device_name = nil)
-        filter = device_name || candidate_disk_names
-        devicegraph.disk_devices.select { |d| filter.include?(d.name) }
+        filter = Array(device_name || candidate_disk_names)
```

Comment:
NP: only a matter of taste:
```suggestion
        filter = device_name ? [device_name] : candidate_disk_names
```